mentat/conversation.py

```diff
@@ -52,7 +55,25 @@ def get_model_response(self, config: ConfigManager) -> (str, list[CodeChange]):
         messages = self.messages.copy()
 
         code_message = self.code_file_manager.get_code_message()
-        messages.append({"role": "system", "content": code_message})
+        system_message = code_message
+
+        if self.code_map:
+            code_map_message = self.code_map.get_message()
```

Comment:
We aren't using the `token_limit` arg here, but if we did we could get the nice fallback behavior you've implemented (trying to return smaller maps if it doesn't fit in the limit). So shouldn't we count tokens first and tell it how much space it has? (right now if I understand correctly it's always using the limit of 2048 tokens set when `CodeMap` is initialized in `app.py`)

Reply:
>We aren't using the token_limit arg here, but if we did we could get the nice fallback behavior you've implemented (trying to return smaller maps if it doesn't fit in the limit)

We're still getting the fallback behavior with or without setting `token_limit` for `self.code_map.get_message()`. Setting `token_limit` in `.get_message()` is just a manual override to whatever `token_limit` the `CodeMap` was instantiated with. We don't *need* it atm, so I can remove it if you'd like. It's useful for development/testing tho.

> So shouldn't we count tokens first and tell it how much space it has?

Could you elaborate on this? I think I'm misinterpreting the question.

Reply:
`token_limit` is just looking at the token count for the generated code map btw

Reply:
Ah gotcha, thanks for clarifying and good call-out :)